extensions/spark/kyuubi-spark-connector-yarn/src/main/scala/org/apache/kyuubi/spark/connector/yarn/YarnLogScan.scala

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.spark.connector.yarn
+
+import scala.collection.mutable
+
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.{FileStatus, FileSystem, Path}
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.connector.read._
+import org.apache.spark.sql.sources.{EqualTo, Filter}
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.sql.util.CaseInsensitiveStringMap
+
+case class YarnLogScan(
+    options: CaseInsensitiveStringMap,
+    schema: StructType,
+    filters: Array[Filter])
+  extends BasicScan {
+  override def readSchema(): StructType = schema
+
+  private val remoteAppLogDirKeyInYarnSite = "yarn.nodemanager.remote-app-log-dir"
+  private val remoteAppLogDirKey = "spark.sql.catalog.yarn.log.dir"
+
+  private val remoteAppLogDir = {
+    val dir = SparkSession.active.sparkContext
+      .getConf.getOption(remoteAppLogDirKey) match {
```

Comment:
this is incorrect.

use `SparkSession.conf` in SQL cases instead of `SparkContext.getConf` in SQL cases because the latter returns the global static conf, which is immutable after Spark is launched.

and I think it is not worth having an additional spark conf key, just use `SparkSession.sessionState.newHadoopConf` to create a `Configuration`, and read the hadoop conf directly.